executor/executor.cc

```diff
@@ -1011,9 +1011,13 @@ void write_coverage_signal(cover_t* cov, uint32* signal_count_pos, uint32* cover
 		bool prev_filter = true;
 		for (uint32 i = 0; i < cov->size; i++) {
 			cover_data_t pc = cover_data[i] + cov->pc_offset;
-			uint32 sig = pc;
-			if (use_cover_edges(pc))
-				sig ^= hash(prev_pc);
+			uint32 sig = pc & 0xFFFFF000;
+			if (use_cover_edges(pc)) {
+				// Only hash the lower 12 bits so the hash is
+				// independent of any module offsets.
+				prev_pc = prev_pc & 0xFFF << 16;
+				sig |= hash((pc & 0xFFF) & prev_pc) & 0xFFF;
```

Comment:
Ah, I think I know why the experiment showed so many signal collisions. This should be something like:

```
sig |= (pc & 0xFFF)^(hash(prev_pc & 0xFFF) & 0xFFF);
```

Reply:
> I'll add a flag to differentiate between the 0 modules/modules cases as well.

Let's skip it for now. After you fix the code, I can run another experiment and we'll see whether we really need that or we can just do the 12 bit manipulation in all cases.

Reply:
Great, thank you! I have fixed the code with the recommended executor change